### ISSUE TITLE
Questions about 6-bit quantization

### ISSUE THREAD

**Issue by Wotoosh** [ChangyuanWang17/QVLM#10]
Dear author，it seems that the code only supports 4-bit quantization using the QVLM and the 8-bit quantization using the LLM.int8(), but I find that results of the 6-bit quantization has been reported in the paper. Is it possible to release the relevant code?Many thanks!

**Comment by ChangyuanWang17** [comment]
Thank you for your attention to our work! As 6-bit quantization is not hardware-friendly, we utilized the pseudo-quantization code from https://github.com/ChangyuanWang17/APQ-DM for the 6-bit experiments.

**Comment by Wotoosh** [comment]
Dear author, after using the code for weight quantization from https://github.com/ChangyuanWang17/APQ-DM/ the spped of calibration and evaluation slows a lot, costs 4 times more than before, is it normal?Many thanks!

**Comment by ChangyuanWang17** [comment]
I'm very pleased to see that you conducted an actual test! However, we must acknowledge that 6-bit is not hardware-friendly. In binary hardware systems, there are no algorithms that can directly accelerate it, and support for 6-bit matrix multiplication is also weak. As a result, pseudo-quantization is required for accuracy-performance testing. Therefore, we recommend using 8-bit and 4-bit settings, which can achieve better practical acceleration. 

That being said, in our tests, using 6-bit quantization took approximately 1.5 times the usual time. May I ask if you are using the weight quantization algorithm from the following link?  
[https://github.com/ChangyuanWang17/APQ-DM/blob/53c1780db8ed453c329baf5d012204f6991e2e4f/utils/quant_util.py#L284](https://github.com/ChangyuanWang17/APQ-DM/blob/53c1780db8ed453c329baf5d012204f6991e2e4f/utils/quant_util.py#L284)

**Comment by Wotoosh** [comment]
> I'm very pleased to see that you conducted an actual test! However, we must acknowledge that 6-bit is not hardware-friendly. In binary hardware systems, there are no algorithms that can directly accelerate it, and support for 6-bit matrix multiplication is also weak. As a result, pseudo-quantization is required for accuracy-performance testing. Therefore, we recommend using 8-bit and 4-bit settings, which can achieve better practical acceleration.
> 
> That being said, in our tests, using 6-bit quantization took approximately 1.5 times the usual time. May I ask if you are using the weight quantization algorithm from the following link? https://github.com/ChangyuanWang17/APQ-DM/blob/53c1780db8ed453c329baf5d012204f6991e2e4f/utils/quant_util.py#L284

Yes I did, and there is something I don't understand. Isn't the quantized weight be fixed after the calibration phase, why can't I use the quantized weight to replace the original weight to avoid the calculation of weight quantization in evaluation phase.

**Comment by Wotoosh** [comment]
What's more,no matter under 6bit scenario or the original 4 bit real quantization ,the test result of scienceqa on the finetuned LLaVA-7B I reproduced is always below than what's reported in the paper.I got 85.24% for 6-bit quantization and 73.36% for 4-bit quantization, I tried for bit quantization again but got a result of 73.05% again.

**Comment by ChangyuanWang17** [comment]
Thank you again for your attention to our work! The results we generated are similar to those reported in the paper. Is the fine-tuned LLaVA-7B model you mentioned the one we open-sourced? You can provide the generated JSONL files, experimental settings, or verification code for further discussion.
